pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamMessageDecoder.java

```diff
@@ -52,7 +52,8 @@ void init(Map<String, String> props, Set<String> fieldsToRead, String topicName)
    * Decodes a row.
    *
    * @param payload The buffer from which to read the row.
-   * @return A new row decoded from the buffer
+   * @return A new row decoded from the buffer. If the returned value is <code>null</code> the row is dropped from the
+   *         segment.
    */
   GenericRow decode(T payload, GenericRow destination);
```

Comment:
Annotate the return as `nullable`
```suggestion
  @Nullable
  GenericRow decode(T payload, GenericRow destination);
```